### PR TITLE
Fix inline datums

### LINKED ISSUE #2835: getData is not abstracted for Babbage
The Alonzo function `collectTwoPhaseScriptInputs` uses the function `getData` to attach datums in outputs to the information needed to run Plutus scripts. The babbage era calls `collectTwoPhaseScriptInputs`, but was supposed to use [getDatum](https://github.com/input-output-hk/cardano-ledger/blob/3510613ed1340d996ba942e525624c8f23118a65/eras/babbage/impl/src/Cardano/Ledger/Babbage/Scripts.hs#L47).

Make `getData` a class method of `ExtendedUTxO` so that it has the correct behavior in each era.

### PR DESCRIPTION
**Pull request by lehins**
Fixes #2835

### CODE REVIEW

File: eras/babbage/impl/src/Cardano/Ledger/Babbage/Scripts.hs [hunk 1]
```diff
@@ -21,46 +22,48 @@ import Cardano.Ledger.Alonzo.TxWitness (TxWitness, unTxDats)
...
+  Maybe (Data era)
+getDatumBabbage tx (UTxO m) sp = do
+  txIn <- getSpendingTxIn sp
+  txOut <- Map.lookup txIn m
+  let txOutDataFromWits = do
+        hash <- txOutDataHash txOut
+        Map.lookup hash (unTxDats (txdats' (getField @"wits" tx)))
+  txOutData txOut <|> txOutDataFromWits
+
+-- TxOut _ _ datum _refScript <- Map.lookup txin m
```
Comment (JaredCorduan) [inline eras/babbage/impl/src/Cardano/Ledger/Babbage/Scripts.hs]: dead code?

File: libs/cardano-ledger-test/src/Test/Cardano/Ledger/Examples/BabbageFeatures.hs [hunk 2]
```diff
@@ -126,16 +126,31 @@ collateralOutput pf =
...
+
+datumExampleOdd :: Data era
+datumExampleOdd = Data (Plutus.I 3)
 
 inlineDatumOutput :: forall era. (Scriptic era) => Proof era -> Core.TxOut era
 inlineDatumOutput pf =
   newTxOut
     pf
-    [ Address (scriptAddr pf (alwaysAlt 3 pf)),
+    [ Address (scriptAddr pf (evenData3ArgsScript pf)),
```
Comment (JaredCorduan) [inline libs/cardano-ledger-test/src/Test/Cardano/Ledger/Examples/BabbageFeatures.hs:142]: awesome, thanks for improving the test!

File: libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/Properties.hs [hunk 3]
```diff
@@ -131,24 +124,21 @@ testTxValidForLEDGER ::
   Proof era ->
   Box era ->
   Property
-testTxValidForLEDGER proof (Box _ (trc@(TRC (_, ledgerState, vtx))) _genstate) =
-  ( if False
-      then trace (show (txSummary proof vtx))
-      else id
-  )
```
Comment (JaredCorduan) [inline libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/Properties.hs:138]: :broom: thanks for cleaning things up as you go!

**Review by JaredCorduan — APPROVED** [review]
Looks great, and as always, I appreciate all the cleanup as well.

File: eras/alonzo/test-suite/src/Test/Cardano/Ledger/Alonzo/PlutusScripts.hs [hunk 4]
```diff
@@ -17,9 +17,6 @@ testingCostModelV2 =
   fromRight (error "testingCostModelV2 is not well-formed") $
     mkCostModel PlutusV2 PV1.costModelParamsForTesting -- TODO use PV2 when it exists
 
-{-# DEPRECATED defaultCostModel "use testingCostModelV1 instead" #-}
-defaultCostModel :: Maybe CostModel
-defaultCostModel = Just testingCostModelV1
```
Comment (JaredCorduan) [inline eras/alonzo/test-suite/src/Test/Cardano/Ledger/Alonzo/PlutusScripts.hs]:
```suggestion
```

to appease ormolu

File: eras/babbage/impl/src/Cardano/Ledger/Babbage/Scripts.hs (hunk 1 above)
Comment (lehins) [reply eras/babbage/impl/src/Cardano/Ledger/Babbage/Scripts.hs]: Yep, it surely was. Thank you for catching it.

**Pull request merged**